[CI:DOCS] `CONTRIBUTING.md`: Update contact information

AndASM opened the pull request:
<!--
Thanks for sending a pull request!

Please make sure you've read and understood our contributing guidelines
(https://github.com/containers/buildah/blob/main/CONTRIBUTING.md) as well as ensuring
that all your commits are signed with `git commit -s`.
-->

#### What type of PR is this?

<!--
Please label this pull request according to what type of issue you are
addressing, especially if this is a release targeted pull request.

Uncomment only one `/kind <>` line, hit enter to put that in a new line, and
remove leading whitespace from that line:
-->
/kind documentation

#### What this PR does / why we need it:
`CONTRIBUTING.md` is outdated. New contributors will be directed to a dead freenode channel.

Changes:
- Freenode is no longer used, IRC has moved to [libera.chat](https://libera.chat).
- \[matrix\] is federated into the `#podman` channel and can be used instead of IRC.

#### How to verify it
Read `CONTRIBUTING.md`.

#### Which issue(s) this PR fixes:

<!--
Automatically closes linked issue when PR is merged.
Uncomment the following comment block and include the issue
number or None on one line.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`, or `None`.
-->

Fixes #4308

#### Special notes for your reviewer:
I am assuming this counts as "developer-facing" not "user-facing" so I have put `None` under the release notes. Is that correct?

#### Does this PR introduce a user-facing change?

<!--
If no, just write `None` in the release-note block below. If yes, a release note
is required: Enter your extended release note in the block below. If the PR
requires additional action from users switching to the new release, include the
string "action required".

For more information on release notes please follow the kubernetes model:
https://git.k8s.io/community/contributors/guide/release-notes.md
-->

```release-note
None
```

rhatdan commented:
LGTM
@TomSweeneyRedHat PTAL

TomSweeneyRedHat commented on CONTRIBUTING.md:
```diff
@@ -143,14 +143,11 @@ is available, showing all PRs awaiting review and approval.
-### For discussions around issues/bugs and features:
-
-#### Buildah Mailing List
+Alternatively [\[matrix\]](https://matrix.org) can be used to access the same channel via federiation at https://matrix.to/#/#podman:chat.fedoraproject.org.
```
I'm 99% sure it should be federation and not federation.

```suggestion
Alternatively, [\[matrix\]](https://matrix.org) can be used to access the same channel via federation at https://matrix.to/#/#podman:chat.fedoraproject.org.
```

TomSweeneyRedHat commented on CONTRIBUTING.md:
```diff
@@ -159,6 +156,12 @@ and
 
+#### Buildah Mailing List
+
+_This list sees low traffic, the GitHub links or IRC above are better options in most cases._
```
```suggestion
```

TomSweeneyRedHat commented on CONTRIBUTING.md:
```diff
@@ -143,14 +143,11 @@ is available, showing all PRs awaiting review and approval.
+### For discussions around issues/bugs and features:
 
 #### GitHub
 You can also use the github
```
I know it's old, but please touch up
```suggestion
You can also use GitHub
```

TomSweeneyRedHat replied:
Also, please add:
`[Discussions](https://github.com/containers/podman/discussions),`

rhatdan commented:
@AndASM Are you still working on this, or would you rather us finish it?

AndASM commented:
> @AndASM Are you still working on this, or would you rather us finish it?

Sorry about that, I missed the update emails and have been distracted and busy lately. I'll try to take a look tomorrow if you haven't otherwise taken care of it.

flouthoc reviewed (changes requested):
Could you reduce length of your commit title please ?

flouthoc commented:
Oops it was commit  2c06b50cf1635f1504bf83a19fb414a6f8bc99c5 caused by using github rebase , @AndASM Could you squash commits into one ?

rhatdan commented:
@AndASM Are you still working on this one?

AndASM commented:
Oops, sorry about that! Life has been a bit crazy of late.

I squashed the commits into one, and rebased it on the current main branch. Is this still okay to pull? It might help someone to have those updates to the documentation. @rhatdan?

rhatdan commented:
I can't seem to reopen, could you open a new PR with just one commit.
